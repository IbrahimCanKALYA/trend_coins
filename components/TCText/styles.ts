import {StyleSheet } from 'react-native';

import {getAdjustedFontSize} from "@/utils/Typography"


export const styles = StyleSheet.create({
  default: {
    fontSize: getAdjustedFontSize(20),
    lineHeight: getAdjustedFontSize(20),
  },
  largeTitle: {
    fontSize: getAdjustedFontSize(20),
    lineHeight: getAdjustedFontSize(20),
    fontWeight: "700"
  },
  title1: {
    fontSize: getAdjustedFontSize(14),
    lineHeight: getAdjustedFontSize(16.94),
    fontWeight: "600"
  },
  subHead: {
    fontSize: getAdjustedFontSize(12),
    lineHeight: getAdjustedFontSize(14.52),
    fontWeight: "600"
  },
  footNote: {
    fontSize: getAdjustedFontSize(10),
    lineHeight: getAdjustedFontSize(12.1),
    fontWeight: "500"
  },
  defaultSemiBold: {
    fontSize: 16,
    lineHeight: 24,
    fontWeight: '600',
  },
  });