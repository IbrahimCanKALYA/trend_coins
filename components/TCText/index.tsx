import { Text, type TextProps, StyleSheet } from 'react-native';

import {Colors} from "@/constants/Colors"
import { styles } from './styles';

export type TCTextProps = TextProps & {
  type?: keyof typeof styles;
  color?: keyof typeof Colors
};

export function TCText({
  style,
  type = 'default',
  color = "text",
  ...rest
}: TCTextProps) {

  return (
    <Text
      style={[styles[type], {color: Colors[color]}, style]}
      {...rest}
    />
  );
}


