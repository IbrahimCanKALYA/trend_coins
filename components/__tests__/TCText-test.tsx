import * as React from 'react';
import renderer from 'react-test-renderer';

import { TCText } from '../TCText';

it(`renders correctly`, () => {
  const tree = renderer.create(<TCText>Snapshot test!</TCText>).toJSON();

  expect(tree).toMatchSnapshot();
});
