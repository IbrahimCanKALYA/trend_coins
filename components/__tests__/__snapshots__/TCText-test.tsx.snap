// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  style={
    [
      {
        "fontSize": 11.250000000000002,
        "lineHeight": 11.250000000000002,
      },
      {
        "color": "#11181C",
      },
      undefined,
    ]
  }
>
  Snapshot test!
</Text>
`;
